tests/src/test/scala/org/http4s/SchemeSpec.scala

```diff
@@ -0,0 +1,61 @@
+package org.http4s
+
+import cats.kernel.laws.OrderLaws
+import org.http4s.internal.parboiled2.CharPredicate
+import org.http4s.testing.HttpCodecTests
+import org.http4s.util.Renderer
+
+class SchemeSpec extends Http4sSpec {
+  "equals" should {
+    "be consistent with equalsIgnoreCase of the values" in {
+      prop { (a: Scheme, b: Scheme) =>
+        (a == b) must_== a.value.equalsIgnoreCase(b.value)
+      }
+    }
+  }
+
+  "compareTo" should {
+    "be consistent with value.compareToIgnoreCase" in {
+      prop { (a: Scheme, b: Scheme) =>
+        a.value.compareToIgnoreCase(b.value) must_== a.compareTo(b)
+      }
+    }
+  }
+
+  "hashCode" should {
+    "be consistent with equality" in {
+      prop { (a: Scheme, b: Scheme) =>
+        (a == b) ==> (a.## must_== b.##)
```

Comment:
Hash code equality is the behavior we want to see? What do we gain through this check?

Reply:
Equality implies hash equality.  The most visible symptom when this rule is broken is use in hash maps: equal objects can go into different buckets and break the invariants of the set.  (Hash equality does not imply equality, because a type may have more than `Int.MaxValue` inhabitants.)

Reply:
I think this is a good test, many times this is forgotten when overriding `equals` and `hashCode`